k2/csrc/determinize.h

```diff
@@ -766,8 +756,10 @@ class DetStateMap {
                          const Fsa &fsa, std::pair<uint64_t, uint64_t> *vec) {
     assert(d.normalized);
 
-    uint64_t a = d.state_id + 17489 * d.seq_len,
-             b = d.state_id * 103979 + d.seq_len;
+    int32_t input_state_id = d.elements.begin()->first;
+
+    uint64_t a = input_state_id + 17489 * d.seq_len,
```

Comment:
Remove input_state_id here, just have 17489 * d.seq_len and so on... and I'll say below..